usr/lib/common/object.c

```diff
@@ -220,9 +221,22 @@ CK_RV object_copy(STDLL_TokData_t * tokdata,
         TRACE_DEVEL("template_validate_attributes failed.\n");
         goto error;
     }
+
+    /*
+     * Let the token know that attributes of the copied object have been
+     * changed.
+     */
+    if (token_specific.t_set_attribute_values != NULL) {
+        rc = token_specific.t_set_attribute_values(tokdata, o, new_tmpl);
+        if (rc != CKR_OK) {
+            TRACE_DEVEL("token_specific_set_attribute_values failed.\n");
```

Comment:
I would also print out the rc value in the trace.

Reply:
yes, will add this.